src/core/unit.h

```diff
@@ -293,6 +293,7 @@ typedef struct Unit {
 
         /* Counterparts in the cgroup filesystem */
         char *cgroup_path;
+        uint64_t cgroup_id;
```

Comment:
hmm, is the cgroup id 0 an invalid cgroup id? If not we should probably initialize this to UINT64_MAX or so, as a marker for "doesn't apply/don't know" instead of zero.

Reply:
From a cursory look at the kernel, [cgroup IDs](https://github.com/torvalds/linux/blob/v5.14/include/linux/cgroup.h#L310-L313) are [inodes](https://github.com/torvalds/linux/blob/v5.14/include/linux/kernfs.h#L150-L154) which AFAIK can't be 0. I also found some [other reference](https://github.com/torvalds/linux/blob/v5.14/tools/testing/selftests/bpf/cgroup_helpers.c#L242-L243) saying that 0 is an invalid cgroup id, but it's just a comment.

Given this I think 0 is fine but I could change it to UINT64_MAX to be safe. Wdyt?

Reply:
we usually use _MAX for "invalid"